crates/bevy_ui/src/render/gradient.rs

```diff
@@ -0,0 +1,984 @@
+                            if point.is_nan() {
+                                let start = i;
+                                let mut end = i + 1;
+                                while end < stops_out.len() - 1 && stops_out[end].1.is_nan() {
+                                    end += 1;
+                                }
+                                let start_point = stops_out[start - 1].1;
+                                let end_point = stops_out[end].1;
+                                let steps = end - start;
+                                let step = (end_point - start_point) / (steps + 1) as f32;
+                                for j in 0..steps {
+                                    stops_out[i + j].1 = start_point + step * (j + 1) as f32;
+                                }
+                                i = end;
+                            }
+                            i += 1;
+                        }
+
+                        extracted_gradients.items.push(ExtractedGradient {
+                            render_entity: commands.spawn(TemporaryRenderEntity).id(),
+                            stack_index: uinode.stack_index,
+                            transform: transform.compute_matrix(),
+                            stops_range: range_start..extracted_color_stops.0.len(),
+                            rect: Rect {
+                                min: Vec2::ZERO,
+                                max: uinode.size,
+                            },
+                            clip: clip.map(|clip| clip.clip),
+                            extracted_camera_entity,
+                            main_entity: entity.into(),
+                            node_type,
+                            border_radius: uinode.border_radius,
+                            border: uinode.border,
+                            resolved_gradient: ResolvedGradient::Conic { center: g_start },
+                        });
+                    }
+                }
+            }
+        }
+    }
+}
+
+#[expect(
+    clippy::too_many_arguments,
+    reason = "it's a system that needs a lot of them"
+)]
+pub fn queue_gradient(
+    extracted_gradients: ResMut<ExtractedGradients>,
+    gradients_pipeline: Res<GradientPipeline>,
+    mut pipelines: ResMut<SpecializedRenderPipelines<GradientPipeline>>,
+    mut transparent_render_phases: ResMut<ViewSortedRenderPhases<TransparentUi>>,
+    mut render_views: Query<(&UiCameraView, Option<&UiAntiAlias>), With<ExtractedView>>,
+    camera_views: Query<&ExtractedView>,
+    pipeline_cache: Res<PipelineCache>,
+    draw_functions: Res<DrawFunctions<TransparentUi>>,
+) {
+    let draw_function = draw_functions.read().id::<DrawGradientFns>();
+    for (index, gradient) in extracted_gradients.items.iter().enumerate() {
+        let Ok((default_camera_view, ui_anti_alias)) =
+            render_views.get_mut(gradient.extracted_camera_entity)
+        else {
+            continue;
+        };
```

Comment:
i think fallible systems let you bubble these now with `?`

Reply:
`?` would return from the system and not just continue though wouldn't it? Or am I misunderstanding

Reply:
true my bad